analyzers/src/SonarAnalyzer.CSharp/Rules/StaticFieldInitializerOrder.cs

```diff
@@ -36,108 +36,64 @@ public sealed class StaticFieldInitializerOrder : SonarDiagnosticAnalyzer
-            identifierFieldMappings
-                .Select(i => new IdentifierTypeDeclarationMapping
-                {
-                    Identifier = i,
-                    TypeDeclaration = GetTypeDeclaration(i.Field)
-                })
-                .Where(mapping => mapping.TypeDeclaration != null)
-                .ToList();
-
-        private static IEnumerable<IdentifierFieldMapping> GetIdentifierFieldMappings(VariableDeclaratorSyntax variable, INamedTypeSymbol containingType, SemanticModel semanticModel) =>
-            variable.Initializer.DescendantNodes()
-                .OfType<IdentifierNameSyntax>()
-                .Select(identifier =>
-                {
-                    if (!containingType.MemberNames.Contains(identifier.Identifier.ValueText))
-                    {
-                        return new IdentifierFieldMapping();
-                    }
-
-                    var field = semanticModel.GetSymbolInfo(identifier).Symbol as IFieldSymbol;
-
-                    return new IdentifierFieldMapping
-                    {
-                        Field = field,
-                        IsRelevant = field is { IsConst: false, IsStatic: true }
-                                     && containingType.Equals(field.ContainingType)
-                                     && semanticModel.GetEnclosingSymbol(identifier.SpanStart) is IFieldSymbol enclosingSymbol
-                                     && enclosingSymbol.ContainingType.Equals(field.ContainingType)
-                    };
-                })
-                .Where(identifier => identifier.IsRelevant);
-
-        private static TypeDeclarationSyntax GetTypeDeclaration(IFieldSymbol field)
+        private static IEnumerable<IFieldSymbol> IdentifierFields(VariableDeclaratorSyntax variable, INamedTypeSymbol containingType, SemanticModel semanticModel)
         {
-            var reference = field.DeclaringSyntaxReferences.FirstOrDefault();
-            return reference?.SyntaxTree == null
-                ? null
-                : reference.GetSyntax().FirstAncestorOrSelf<TypeDeclarationSyntax>();
-        }
-
-        private struct IdentifierFieldMapping
-        {
-            public IFieldSymbol Field { get; set; }
-
-            public bool IsRelevant { get; set; }
+            foreach (var identifier in variable.Initializer.DescendantNodes().OfType<IdentifierNameSyntax>())
+            {
+                if (containingType.MemberNames.Contains(identifier.Identifier.ValueText)
+                    && semanticModel.GetSymbolInfo(identifier).Symbol is IFieldSymbol { IsConst: false, IsStatic: true } field
+                    && containingType.Equals(field.ContainingType)
+                    && semanticModel.GetEnclosingSymbol(identifier.SpanStart) is IFieldSymbol enclosingSymbol
+                    && enclosingSymbol.ContainingType.Equals(field.ContainingType))
+                {
+                    yield return field;
```

Comment:
Consider turning this into an expression-bodied Linq query

Reply:
I tried, it was ugly